pkg/kubelet/kubelet_pods.go

```diff
@@ -320,7 +308,7 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 			}
 		}
 
-		if subPath != "" {
+		if subPath != "" && hostPath != "" {
```

Comment:
If we want to support the subpath feature for image volume, we can change it like this in the kubelet after it is merged:

```go
var image *runtimeapi.ImageSpec
var imageSubPath string
if imageVolumes != nil && utilfeature.DefaultFeatureGate.Enabled(features.ImageVolume) {
    image = imageVolumes[mount.Name]
    imageSubPath = subPath
}
```

Reply:
> if subPath != "" && hostPath != "" {

We don't need to check the volume type because all traditional volume plugins always return a non-empty host path.